applications/DEMApplication/custom_constitutive/DEM_D_Linear_Simple_Coulomb_CL.cpp

```diff
@@ -0,0 +1,131 @@
+                                                    double ViscoDampingLocalContactForce[3],
+                                                    double& cohesive_force,
+                                                    SphericParticle* element1,
+                                                    SphericParticle* element2,
+                                                    bool& sliding,
+                                                    double LocalCoordSystem[3][3]) {
+    // Compute stiffness coefficients
+    const double my_radius    = element1->GetRadius();
+    const double other_radius = element2->GetRadius();
+    const double equiv_radius = 2.0 * my_radius * other_radius / (my_radius + other_radius);
+    const double my_young     = element1->GetYoung();
+    const double my_poisson   = element1->GetPoisson();
+    
+    mKn = my_young * equiv_radius; // normal
+    mKt = my_poisson * mKn;        // tangent
+
+    // Compute normal and tangent forces
+    const double normal_contact_force = mKn * indentation;
+    LocalElasticContactForce[2] = normal_contact_force;
+
+    CalculateTangentialForceWithNeighbour(normal_contact_force, OldLocalElasticContactForce, LocalElasticContactForce, LocalDeltDisp, sliding, element1, element2);
+  }
+
+  void DEM_D_Linear_Simple_Coulomb::CalculateForcesWithFEM(const ProcessInfo& r_process_info,
+                                                           const double OldLocalElasticContactForce[3],
+                                                           double LocalElasticContactForce[3],
+                                                           double LocalDeltDisp[3],
+                                                           double LocalRelVel[3],
+                                                           double indentation,
+                                                           double previous_indentation,
+                                                           double ViscoDampingLocalContactForce[3],
+                                                           double& cohesive_force,
+                                                           SphericParticle* const element,
+                                                           Condition* const wall,
+                                                           bool& sliding) {
+    // Compute stiffness coefficients
+    const double my_radius  = element->GetRadius();
+    const double my_young   = element->GetYoung();
+    const double my_poisson = element->GetPoisson();
+
+    mKn = my_young * my_radius; // normal
+    mKt = my_poisson * mKn;     // tangent
+
+    // Compute normal and tangent forces
+    const double normal_contact_force = mKn * indentation;
+    LocalElasticContactForce[2] = normal_contact_force;
+
+    CalculateTangentialForceWithNeighbour(normal_contact_force, OldLocalElasticContactForce, LocalElasticContactForce, LocalDeltDisp, sliding, element, wall);
+  }
+
+  template<class NeighbourClassType>
```

Comment:
A template function is suggested to be put in the head file if I am right. Please look at this [link](https://stackoverflow.com/questions/42150072/template-function-body-in-header-or-cpp#:~:text=If%20it's%20a%20template%20method,is%20going%20to%20look%20like.).

Reply:
That's true. Done it.